ui/hooks/useNftCollectionsMetadata.ts

```diff
@@ -0,0 +1,64 @@
+import { useEffect, useMemo, useState } from 'react';
+import { NameType } from '@metamask/name-controller';
+import { Collection } from '@metamask/assets-controllers';
+import { useSelector } from 'react-redux';
+import { getCurrentChainId } from '../selectors';
+import { fetchNftCollectionsMetadata } from '../store/actions';
+
+export type UseNftCollectionsMetadataRequest = {
+  value: string;
+  type: NameType;
+};
+
+type CollectionsData = {
+  [key: string]: Collection;
+};
+
+export function useNftCollectionsMetadata(
+  requests: UseNftCollectionsMetadataRequest[],
+) {
+  const chainId = useSelector(getCurrentChainId);
+  const [collectionsMetadata, setCollectionsMetadata] =
+    useState<CollectionsData>({});
+
+  const memoisedRequestsWithChainId = useMemo(() => {
+    return {
+      contracts: requests
+        .filter(({ value }) => value)
+        .map(({ value }) => value),
+      chainId,
+    };
+  }, [JSON.stringify(requests), chainId]);
```

Comment:
`JSON.stringify` creates a new string every time, so this defeats the purpose of the dependency array. Using `requests` should work.

Reply:
It does create a new string but string is not a reference type so it will actually check the equality before and after with given strings.

If we pass the `requests` here, and want to use `requests` directly, we must guarantee that reference is not changing which we cannot do that since `requests` drops here as a new object evertyime.

In `ui/hooks/useDisplayName.ts` this is what passed to this hook.
```
 const nameRequests = requests.map(({ value, type }) => ({
    value,
    type,
  }));
```

In order to make this work with reference, we need in parent. 
```
 const nameRequests = useMemo(
  () => requests.map(({ value, type }) => ({
    value,
    type,
  }));
  );
```

So this could be another solution. 
What do you think that which one would be better solution?